scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -323,7 +323,8 @@ private int selectAndStartTasks(Policy currentPolicy, Map<JobId, JobDescriptor>
 
         while (progressiveIterator.hasMoreElements() && !freeResources.isEmpty()) {
 
-            LinkedList<EligibleTaskDescriptor> taskRetrievedFromPolicy = new LinkedList<>(progressiveIterator.getNextElements(freeResources.size()));
+            LinkedList<EligibleTaskDescriptor> taskRetrievedFromPolicy = new LinkedList<>(progressiveIterator.getNextElements(PASchedulerProperties.SCHEDULER_POLICY_USE_FREE_NODES.getValueAsBoolean() ? freeResources.size()
+                                                                                                                                                                                                        : Integer.MAX_VALUE));
```

Comment:
@fviale the Integer.MAX_VALUE is scarying. Could it make an issue here in the sublist ? https://github.com/ow2-proactive/scheduling/blob/a47870e681640fa0cff88146fa8a47a5d4ca580b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/helpers/VariableBatchSizeIterator.java#L55

Reply:
Ok I will have a look

Reply:
I made a change in VariableBatchSizeIterator to ensure it does not cause issues